### PR TITLE
lots of changes in this commit. most of the changes are related to

### PR DESCRIPTION
**Pull request by stateoflux**
most of the changes are related to retrieving activity data from the api.  also optimized  some of the
methods in the UserService object and cleaned up logic in the login and
signup controllers.  a user is now able to signup/login and retrieve their
activity reliably.

### CODE REVIEW

File: public/js/src/controllers/cardsController.js [hunk 1]
```diff
@@ -23,6 +23,22 @@ angular.module('fittr.controllers')
...
+  // Retrieve current user's activity for the past 7 days
+  // and store in memory and localStorage
+  UserService.currentUser = UserService.getFromLocal();
+
+  UserService.getActivity(UserService.currentUser._id, 7)
+    .then(function(data) {
+      console.log("response from get activity: ", data);
+      debugger;
+      UserService.saveActivity(UserService.currentUser._id, data);
+      $scope.currentUser = UserService.currenUser;
```
Comment (sarchila) [inline public/js/src/controllers/cardsController.js:35]: UserService.currenUser  -  typo?

File: gruntfile.js [hunk 2]
```diff
@@ -2,6 +2,8 @@
 module.exports = function(grunt) {
   'use strict';
   // Project configuration.
+  var clientSideJavascriptFiles = "public/js/src/**/*.js";
```
Comment (sarchila) [inline gruntfile.js:5]: Nice!

File: public/js/src/services/userService.js [hunk 3]
```diff
@@ -63,40 +65,45 @@ angular.module('fittr.services')
       },
 
       save: function(userData) {
+        delete userData.__v;
+        delete userData.password;
+        delete userData.salt;
```
Comment (sarchila) [inline public/js/src/services/userService.js:70]: Oh cool.  Nice that we're not saving that stuff or passing it around.

File: public/js/src/controllers/cardsController.js (hunk 1 above)
Comment (stateoflux) [inline public/js/src/controllers/cardsController.js:35]: good catch1

Comment (stateoflux) [inline public/js/src/controllers/cardsController.js:35]: catch!

**Pull request merged**